pkg/datapath/iptables/ipset/cell.go

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package ipset
+
+import (
+	"context"
+	"os/exec"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/cilium/cilium/pkg/datapath/tables"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/statedb/reconciler"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+// Cell exposes methods to add and remove node IPs from the kernel IP sets.
+// The sets are in turn referenced by iptables rules to exclude traffic
+// to cluster nodes from being masqueraded.
+// There are two distinct sets, one for IPv4 addresses and one for IPv6
+// addresses.
+// Internally, the cell stores the desired IP sets state in a StateDB table
+// and uses a reconciler to update the realized state (that is, the actual
+// kernel IP sets).
+// Other sets that do not pertain to Cilium configuration are not changed
+// in any way.
+var Cell = cell.Module(
+	"ipset",
+	"Handle kernel IP sets configuration for Cilium",
+
+	cell.Provide(newIPSetManager),
+
+	cell.ProvidePrivate(
+		tables.NewIPSetTable,
+
+		reconciler.New[*tables.IPSet],
+		newReconcilerConfig,
+		newOps,
+	),
+	cell.ProvidePrivate(func(logger logrus.FieldLogger) *ipset {
+		return &ipset{
+			executable: funcExecutable(func(ctx context.Context, name string, arg ...string) ([]byte, error) {
+				return exec.CommandContext(ctx, name, arg...).Output()
+			}),
+			log: logger,
+		}
+	}),
+	cell.ProvidePrivate(func(cfg *option.DaemonConfig) config {
+		return config{NodeIPSetNeeded: cfg.NodeIpsetNeeded()}
```

Comment:
Any reason why we can't inline `cfg.NodeIpsetNeeded()` here in a last commit? It seems this is the only remaining usage

Reply:
It is used here too:

https://github.com/cilium/cilium/blob/e8c781a929d2e8132ae7cce2e6d73718916871d1/pkg/datapath/iptables/cell.go#L31

to let the iptables manager know if it should install the rule to exclude traffic to cluster nodes from being masqueraded:

https://github.com/cilium/cilium/blob/e8c781a929d2e8132ae7cce2e6d73718916871d1/pkg/datapath/iptables/iptables.go#L1196

IOW the ipset reconciler has the responsibility of building and keeping the Cilium IP sets up to date, while the iptables manager is still in charge to reference them to avoid masquerading traffic to those IPs.
Given that the `NodeIpsetNeeded` depends on multiple global options, I left that as a global utility function to be accessed by both modules.